src/document.cpp

```diff
@@ -1669,6 +1673,22 @@ rvoe<CapyPDF_TransparencyGroupId> PdfDocument::add_transparency_group(PdfDrawCon
     return CapyPDF_TransparencyGroupId{(int32_t)transparency_groups.size() - 1};
 }
 
+rvoe<CapyPDF_SoftMaskId> PdfDocument::add_soft_mask(const SoftMask &sm) {
+    std::string subtype = sm.S == CAPY_SOFT_MASK_ALPHA ? "Alpha" : "Luminosity";
```

Comment:
This creates a temporary copy of the string that is then immediately destroyed. The ternary should be directly in the format call below.

Reply:
Fixed.